libs/auth-nest-tools/src/lib/jwt.strategy.ts

```diff
@@ -54,7 +54,7 @@ export class JwtStrategy extends PassportStrategy(Strategy) {
         nationalId: actor.nationalId,
         scope: this.parseScopes(actor.scope),
       },
-      ip: String(request.headers['x-real-ip']) ?? request.ip,
+      ip: String(request.headers['x-forwarded-for'] ?? request.ip),
```

Comment:
The plan was originally to use Express's built in support for parsing X-Forwarded-For:

http://expressjs.com/en/guide/behind-proxies.html

There's already an [environment variable](https://github.com/island-is/island.is/blob/54cb834e0157b87da61932109debdcf1f00ca6db/libs/infra-nest-server/src/lib/bootstrap.ts#L82-L87) set up which we can use to configure this everywhere.

The tricky thing is that all of this needs to also support [forwarding](https://github.com/island-is/island.is/blob/2bf24efe74404baaebcfd53846b52854688cd867/libs/auth-nest-tools/src/lib/auth.middleware.ts#L67) within our infrastructure, which currently uses `x-real-ip`.

We at least need to switch that to set the x-forwarded-for header, but I would really like for us to configure the "trust proxy" value correctly and just use express's calculated `request.ip` here.

Our internal APIs could have `EXPRESS_TRUST_PROXY=1`. But our "public" apps might need some environment-specific value, either the correct number of proxies, or an array with whitelisted IP ranges.

Reply:
I like the whitelist approach but because the cloudfront ips can change it might be troublesome to hardcode them.

The number approach feels a bit fragile, if either the number of ip's would increase or decrease it would break.
If it would decrease the user could enter any ip in the X-Forwarded-For header and we would use it.

I would choose to log them all for the time being, but I know this is not a great solution for the long term.